src/core/SampleBuffer.cpp

```diff
@@ -90,6 +90,12 @@ SampleBuffer::SampleBuffer() :
 SampleBuffer::SampleBuffer(const QString & audioFile, bool isBase64Data)
 	: SampleBuffer()
 {
+	if (auto cachedBuffer = Engine::getSampleBufferCache()->get(audioFile.toStdString()); cachedBuffer.has_value())
```

Comment:
In my opinion the caching and the update of the cache should be done outside of `SampleBuffer`. Things that are cached should not know that they are cached or know anything about the caching mechanism.

The `SampleBuffer` should ideally just be a low level buffer for samples and it should not know the context in which it is used. The fact that the low level object `SampleBuffer` accesses the high level object `Engine` shows that there is likely a problem with an indirection.

Reply:
I was replanning to remove the LRU cache idea anyways, and just make it so that the project has a "Loaded Samples" Sidebar, and then we could just store and share the loaded samples in memory for as many samples the user needs. The user could then manage the cache on their own, so it isn't completely a black box. I should've drafted this earlier in that case.

And I would still need to continue working on #6610 since it would make a lot of what I'm trying to do here easier.